backend/src/test/java/com/google/sps/TestEmailNotifier.java

```diff
@@ -0,0 +1,109 @@
+package com.google.sps;
+
+import com.google.sps.notifs.EmailNotifier;
+import org.junit.Assert;
+import org.junit.Test;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+import java.util.Properties;
+import static org.junit.Assert.*;
+
+public class TestEmailNotifier {
+
+  public MimeMessage createTestMessage() {
+    Properties properties = System.getProperties();
+    Session session = Session.getDefaultInstance(properties);
+    return new MimeMessage(session);
+  }
+
+  private String getTextFromMessage(MimeMessage message) throws MessagingException, IOException {
+    String result = "";
+    result = message.getContent().toString();
+    return result;
+  }
+
+  @Test
+  public void constructorShouldSetNameAndEmailTo() {
+    EmailNotifier emailNotifier = new EmailNotifier("John", "jdoe@gmail.com", null);
+    assertEquals("John", emailNotifier.getRecipientName());
+    assertEquals("jdoe@gmail.com", emailNotifier.getToEmail());
+  }
+
+  @Test
+  public void testMessageHasCorrectApplicationName() throws MessagingException {
+    EmailNotifier emailNotifier = new EmailNotifier("Jordan", "jjgrant@buffalo.edu", null);
+    MimeMessage mimeMessage = emailNotifier.createEmail("", "", "");
+    MimeMessage message = createTestMessage();
+    message.setFrom(new InternetAddress("Adlib-Step@gmail.com"));
+    Assert.assertArrayEquals(
+        "The Application Name should be the same as the global variable in the Ad-lib class",
+        message.getFrom(),
+        mimeMessage.getFrom());
+  }
+
+  @Test
+  public void testMessageHasCorrectSubject() throws MessagingException {
+    EmailNotifier emailNotifier = new EmailNotifier("Jordan", "jjgrant@buffalo.edu", null);
+    MimeMessage mimeMessage = emailNotifier.createEmail("", "Random Subject", "");
+    MimeMessage message = createTestMessage();
+    message.setSubject("Random Subject");
+    Assert.assertEquals(
+        "These two subjects should be the same", message.getSubject(), mimeMessage.getSubject());
+  }
+
+  @Test
+  public void testMessageShouldHaveCorrectBodyText() throws MessagingException, IOException {
+    EmailNotifier emailNotifier = new EmailNotifier("Jordan", "jjgrant@buffalo.edu", null);
+    MimeMessage mimeMessage = emailNotifier.createEmail("", "", "Random Text");
+    MimeMessage message = createTestMessage();
+    message.setText("Random Text");
+    Assert.assertEquals(
+        "These two strings should not be the same ",
+        getTextFromMessage(message),
+        getTextFromMessage(mimeMessage));
+  }
```

Comment:
same suggestion as the previous test method: As much as possible we want to test the intended public API of this class. We should be able capture the emails sent by the mock gmail service and assert that the subject and body are the same